tests/_utils/model.js

```diff
@@ -138,367 +179,223 @@ export function stringify( node, selectionOrPositionOrRange = null ) {
+			if ( !conversionApi.schema.check( schemaQuery ) ) {
+				throw new Error( `Element '${ schemaQuery.name }' not allowed in context.` );
+			}
 
-		if ( tagName[ 0 ] == '/' ) {
-			return {
-				type: 'selectionEnd'
-			};
-		}
+			if ( consumable.consume( data.input, { name: true } ) ) {
+				data.output = new ModelElement( data.input.name, data.input.getAttributes() );
+				conversionApi.mapper.bindElements( data.output, data.input );
 
-		if ( tagExtension.endsWith( ' /' ) ) {
-			return {
-				type: 'collapsedSelection',
-				attributes: parseAttributes( tagExtension.slice( 1, -2 ) )
-			};
+				data.context.push( data.output );
+				data.output.appendChildren( conversionApi.convertChildren( data.input, consumable, data ) );
+				data.context.pop();
+			}
 		}
+	};
+}
 
-		return {
-			type: 'selectionStart',
-			attributes: parseAttributes( tagExtension.slice( 1 ) )
+function convertToModelText( withAttributes = false ) {
+	return ( evt, data, consumable, conversionApi ) => {
+		const schemaQuery = {
+			name: '$text',
+			inside: data.context
 		};
-	},
-
-	tag( match ) {
-		const tagContents = match[ 1 ].split( /\s+/ );
-		const tagName = tagContents.shift();
-		const attrs = tagContents.join( ' ' );
-
-		if ( tagName == '/$text' ) {
-			return {
-				type: 'textEnd'
-			};
-		}
 
-		if ( tagName == '$text' ) {
-			return {
-				type: 'textStart',
-				attributes: parseAttributes( attrs )
-			};
+		if ( !conversionApi.schema.check( schemaQuery ) ) {
+			throw new Error( `Element '${ schemaQuery.name }' not allowed in context.` );
 		}
 
-		if ( tagName[ 0 ] == '/' ) {
-			return {
-				type: 'closingTag',
-				name: tagName.slice( 1 )
-			};
-		}
-
-		return {
-			type: 'openingTag',
-			name: tagName,
-			attributes: parseAttributes( attrs )
-		};
-	},
-
-	text( match ) {
-		return {
-			type: 'text',
-			data: match[ 0 ]
-		};
-	}
-};
+		if ( conversionApi.schema.check( schemaQuery ) ) {
+			if ( consumable.consume( data.input, { name: true } ) ) {
+				let node;
 
-function *tokenize( data ) {
-	while ( data ) {
-		const consumed = consumeNextToken( data );
+				if ( withAttributes ) {
+					node = new ModelText( data.input.getChild( 0 ).data, data.input.getAttributes() );
+				} else {
+					node = new ModelText( data.input.data );
+				}
 
-		data = consumed.data;
-		yield consumed.token;
-	}
+				data.output = node;
+			}
+		}
+	};
 }
 
-function consumeNextToken( data ) {
-	let match;
+// -- converters model -> view -----------------------------------------------------
 
-	for ( let patternName in patterns ) {
-		match = data.match( patterns[ patternName ] );
+function insertElement() {
+	return ( evt, data, consumable, conversionApi ) => {
+		consumable.consume( data.item, 'insert' );
 
-		if ( match ) {
-			data = data.slice( match[ 0 ].length );
+		const viewPosition = conversionApi.mapper.toViewPosition( data.range.start );
+		const viewElement = new ViewElement( data.item.name, data.item.getAttributes() );
```

Comment:
Use can use generic `insertElement` and add view element constructor as a parameter. Something like this:
```
insertElement( data => new ViewElement( data.item.name, data.item.getAttributes() ) );
```